Adopt RSCSS system

mzdr opened the pull request:
With this pull request I would like to update our CSS guidelines.
## RSCSS

Right now we are using the [BEM](http://getbem.com/) syntax. Over the last couple of projects we encountered some issues with it. One of the biggest was an on going raise of long selectors — which of course can be handled by linting and stricter rules. But nevertheless…

I stumbled upon [RSCSS](http://rscss.io/) during these projects which is a …

> A set of simple ideas to guide your process of building maintainable CSS.
> – [rscss.io](http://rscss.io/)

Since BEM isn't bad at all and RSCSS pretty much follows BEM conventions<sup>[[1](#references)]</sup>, only with a different syntax I thought we can give it a try without changing too much at a time.
## File and folder structure

With that and HTTP2 (#6) in mind I started to update our [gridonic/web-boilerplate](https://github.com/gridonic/web-boilerplate) ([feature/http2](https://github.com/gridonic/web-boilerplate/tree/feature/http2)). Specifically speaking I changed our structure to a more [strict component based](https://github.com/gridonic/web-boilerplate/tree/feature/http2/src/components/) approach. _Now that we are going to stick to the RSCSS system, we can also have our file and folder structure in a more component orientated approach._<sup>[[2](#references)]</sup>

So the major changes are:
- **No** more bundles or modules, **just** components
- **All** components have their own directory within the components folder<sup>[[2](#references)]</sup>
- Each component has **their** styling inside its folder<sup>[[2](#references)]</sup>
- Non-component styling will still be in the (global) app-wide counterparts

I'm just trying to eliminate complexity by simplifying the structure. In my opinion everything on a website **is a** component, even if it's a collection of components. So why not drop any surplus naming and keep it simple?

<a name="references"></a>
## References
1. http://rscss.io/other-resources.html
2. These changes are closely associated with the proposed changes in #6

Commit:
added wip. browser implementation part is missing

Commit:
fixed ToC links

Commit:
updated sass-lint.yml

Commit:
removed http2 related notes

peschee commented:
@gridonic/core Please review and comment where appropriate.

peschee commented on css/README.md:
```diff
@@ -1,364 +1,350 @@
+- Prefix property values or color parameters with a leading zero (e.g., `0.5` instead of `.5` and `-0.5px` instead of `-.5px`).
+- Lowercase all hex values, e.g., `#fff`. Lowercase letters are much easier to discern when scanning a document as they tend to have more unique shapes.
+- Use shorthand hex values where available, e.g., `#fff` instead of `#ffffff`.
+- Quote attribute values in selectors, e.g., `input[type="text"]`. [They’re only optional in some cases](http://mathiasbynens.be/notes/unquoted-attribute-values#css), and it’s a good practice for consistency.
```
Yes! 👍

peschee commented on css/README.md:
```diff
@@ -1,364 +1,350 @@
-.cell--detail,
-.cell--important {
-    border-width: 4px;
+**Use no more than 1 level of nesting!**
```
I agree, however, there are some exceptions where I think nesting up to 2 levels can be more readable, while still yielding the same code. Example:

``` sass
.nav {
    > .item {
        &::after {
            // some styles
        }

        &:last-of-type {
            // some styles
        }
    }
}
```

versus (when maximum nesting level 1 is forced)

``` sass
.nav {
    > .item {
        // some styles
    }

    > .item::after {
        // some styles
    }

    > .item:last-of-type {
        // some styles
    }
}
```

Can we somehow accommodate this? What do you guys think @gridonic/core ?

mzdr replied:
Yep you're right! Let's keep it to a max-depth of 2.

Commit:
changed sass-lint max-depth to 2

peschee commented:
Any other takers @gridonic/core ?

Commit:
removed alphabetically ordering requirement

mzdr commented:
I also removed the alphabetically ordering statement which we discussed last time as this is a little bit too specific.

Commit:
Exclude -webkit-overflow-scrolling 

...from misspelled properties

@see https://github.com/sasstools/sass-lint/issues/693

Commit:
Codestyle fix

peschee commented on css/README.md:
```diff
@@ -1,364 +1,350 @@
-## Comments
+### Ordering of property declarations
 
-* Keep line-length to a sensible maximum, e.g., 80 columns.
```
@mzdr Can we agree on a bigger value here? 80 is very old and was mainly introduced because of the small screens. However, it is quite little on today's screens. How about 120?

mzdr replied:
Sure, 120 is fine for me! I didn't really think about the 80 chars limit and where it does come from. I just stumbled upon this: [Why is 80 characters the 'standard' limit for code width?](https://programmers.stackexchange.com/questions/148677/why-is-80-characters-the-standard-limit-for-code-width)

Pretty funny, the IBM punch card.

peschee replied:
Added the change

peschee commented:
Have looked over it, thank you very much. To me it looks awesome and we should definitely merge this! I have added some comments along some lines. Also, I have directly made two commits with corrections.

Commit:
Use 120 instead of 80 chars for the number of columns

https://programmers.stackexchange.com/questions/148677/why-is-80-characters-the-standard-limit-for-code-width is old, I guess we can accommodate modern screen sizes better.
